File: Assets/scenes/landinglaunchscene/script/JumpNode.js
#pragma strict

//var destinationScene : int;
var jumpNodeFrequency : int;
var forcedFail : boolean; //do we force a failure on this jump?


var gateEnabled : boolean = true;


var test : boolean = false;



private var theShip : GameObject;
private var oscSender : OSCSystem;



function Start () {
	theShip = GameObject.Find("TheShip");
	oscSender = GameObject.Find("PersistentScripts").GetComponent.<OSCSystem>();
}


function OnTriggerEnter (other : Collider) {
	if(gateEnabled){
		if (other.gameObject == theShip){
			theShip.GetComponent.<JumpSystem>().inGate = true;
			
			theShip.GetComponent.<JumpSystem>().updateJumpStatus();
			
			var ps : PersistentScene = GameObject.Find("PersistentScripts").GetComponent.<PersistentScene>();
			
			ps.forcedHyperspaceFail = forcedFail;
			
			
			
		} 
	}
}

function explode(){
	//stop rotation
	var rr : RingRotationBehaviour[] = GetComponentsInChildren.<RingRotationBehaviour>();
	for (var p in rr){
		p.speed = 0;
	}
	
	var bf : BlinkenFlareBehaviour[] = GetComponentsInChildren.<BlinkenFlareBehaviour>();
	for (var b in bf){
		b.blinking = false;
		b.flickerAndDie();
	}
	
	//disable collider
	var c : Collider = GetComponentInChildren.<Collider>();
	c.enabled = false;
	
	var part : ParticleSystem = GetComponentInChildren.<ParticleSystem>();
	part.enableEmission = false;
	
	
	
	for(var t : Transform in GetComponentsInChildren.<Transform>()){
		Debug.Log(t.name);
		if(t.name.Contains("Plane")){
			t.parent = null;
			t.gameObject.AddComponent.<Rigidbody>();
			t.rigidbody.useGravity = false;
			t.rigidbody.AddForce( (transform.position - t.position).normalized * -10, ForceMode.Impulse);
			t.rigidbody.AddTorque(Vector3(0,40,0), ForceMode.Impulse);
			
			
		}
	}
	
	
	
}
	


function OnTriggerExit (other : Collider) {
	
	if (other.gameObject == theShip){
		theShip.GetComponent.<JumpSystem>().inGate = false;
	    theShip.GetComponent.<JumpSystem>().updateJumpStatus();
		
	} 
}
function Update () {
	if(test){
		test = false;
		explode();
	}
	//transform.rotation *= Quaternion.Euler(0.01,0,0);
}

File: Assets/scenes/landinglaunchscene/script/JumpNode.cs
using UnityEngine;
using System;


public class JumpNode:MonoBehaviour{
	
	//var destinationScene : int;
	public int jumpNodeFrequency;
	public bool forcedFail; //do we force a failure on this jump?
	
	
	public bool gateEnabled = true;
	
	
	public bool test = false;
	
	
	
	GameObject theShip;
	OSCSystem oscSender;
	
	
	
	public void Start() {
		theShip = GameObject.Find("TheShip");
		oscSender = GameObject.Find("PersistentScripts").GetComponent<OSCSystem>();
	}
	
	
	public void OnTriggerEnter(Collider other) {
		if(gateEnabled){
			if (other.gameObject == theShip){
				theShip.GetComponent<JumpSystem>().setFlatSpace(true);
				

				PersistentScene ps = GameObject.Find("PersistentScripts").GetComponent<PersistentScene>();
				

			} 
		}
	}
	
	public void explode(){
		//stop rotation
		RingRotationBehaviour[] rr = GetComponentsInChildren<RingRotationBehaviour>();
		foreach(RingRotationBehaviour p in rr){
			p.speed = 0.0f;
		}
		
		BlinkenFlareBehaviour[] bf = GetComponentsInChildren<BlinkenFlareBehaviour>();
		foreach(BlinkenFlareBehaviour b in bf){
			b.blinking = false;
			b.flickerAndDie();
		}
		
		//disable collider
		Collider c = GetComponentInChildren<Collider>();
		c.enabled = false;
		
		ParticleSystem part = GetComponentInChildren<ParticleSystem>();
		part.enableEmission = false;
		
		
		
		foreach(Transform t in GetComponentsInChildren<Transform>()){
			UnityEngine.Debug.Log(t.name);
			if(t.name.Contains("Plane")){
				t.parent = null;
				t.gameObject.AddComponent<Rigidbody>();
				t.GetComponent<Rigidbody>().useGravity = false;
				t.GetComponent<Rigidbody>().AddForce( (transform.position - t.position).normalized * -10, ForceMode.Impulse);
				t.GetComponent<Rigidbody>().AddTorque(new Vector3(0.0f,40.0f,0.0f), ForceMode.Impulse);
				
				
			}
		}
		
		
		
	}
		
	
	
	public void OnTriggerExit(Collider other) {
		
		if (other.gameObject == theShip){
			theShip.GetComponent<JumpSystem>().setFlatSpace(false);
			
		} 
	}
	public void Update() {
		if(test){
			test = false;
			explode();
		}
		//transform.rotation *= Quaternion.Euler(0.01,0,0);
	}
}
